indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -162,15 +162,21 @@ default int getPriority()
 
   /**
    * Asks a task to arrange for its "run" method to exit promptly. Tasks that take too long to stop gracefully will be
-   * terminated with extreme prejudice. Note that this method can be called at any time while {@link #run} is called.
-   * Its implementations should handle potential concurreny issues properly.
+   * terminated with extreme prejudice.
+   *
+   * This method can be called at any time while {@link #run} is being called when the task is killed.
```

Comment:
Could you specify what should or should not happen if `stopGracefull()` is called concurrently (e. g. "before") `run()`? Does it guarantee that the task won't even start?

Reply:
Good point. I added the below to javadoc.

> This method can be called at any time while run() is being called when the task is killed. If this task is not started yet, that is run() is not called yet, this method will be never called. Once this task is started, this method can be called even after run() returns. Implementations of this method may want to avoid unnecessary work if run() already returned.